### PR TITLE
Certbot

### PR DESCRIPTION
**Pull request by Sybrand**
- Add friendly domain route (template, bash scripts, jenkins, github actions).
- Add certbot cronjob (prod only).
- Update openshift template version. (was giving error).
- Re-factored cronjob names (for easier cleanup).
- Remove redundant ocp3 zap scan.
- Removed redunant ocp3 mariadb scripts.
- Removed redundant references to ocp3.
- Reduced production cpu request size.

### CODE REVIEW

File: openshift/Jenkinsfile [hunk 1]
```diff
@@ -60,6 +60,8 @@ pipeline {
                     sh "PROJ_TARGET=${projProd} CPU_REQUEST=1000m CPU_LIMIT=2000m ./openshift/scripts/oc_provision_backup_postgres_cronjob.sh prod apply"
                     sh "PROJ_TARGET=${projProd} BACKUP_VOLUME_SIZE=3Gi ./openshift/scripts/oc_provision_backup_mariadb.sh prod apply"
                     sh "PROJ_TARGET=${projProd} ./openshift/scripts/oc_provision_backup_mariadb_cronjob.sh prod apply"
+                    // Configure 
+                    sh "PROJ_TARGET=${projProd} CERTBOT_STAGING=false DRYRUN=false CERTBOT_SERVER=\"https://www.entrust.net/acme/api/v1/directory/98-YMBC-M0BH\" ./openshift/scripts/oc_provision_certbot_cronjob.sh prod apply"
```
Comment (andrea-williams) [inline openshift/Jenkinsfile:64]: what is the `98-YMBC-blahblah` about? Is that something specific to us (and if so, should we be making it a config map/secret variable?

File: openshift/scripts/oc_cleanup.sh [hunk 2]
```diff
@@ -28,40 +28,20 @@ if [ "${APPLY}" ]; then
 else
 	DELETE_OR_GET="get"
 fi
-OC_CLEAN_DEPLOY="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} all,cm -o name -l app=${NAME_OBJ}"
-OC_CLEAN_HOURLY_CRONJOB="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} --ignore-not-found=true cronjob/bcfw-p1-hourly-actuals-${NAME_APP}-${SUFFIX}"
-OC_CLEAN_FORECAST_CRONJOB="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} --ignore-not-found=true cronjob/bcfw-p1-forecasts-${NAME_APP}-${SUFFIX}"
-OC_CLEAN_EC_GDPS_CRONJOB="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} --ignore-not-found=true cronjob/env-canada-gdps-${NAME_APP}-${SUFFIX}"
-OC_CLEAN_EC_HRDPS_CRONJOB="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} --ignore-not-found=true cronjob/env-canada-hrdps-${NAME_APP}-${SUFFIX}"
-OC_CLEAN_EC_RDPS_CRONJOB="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} --ignore-not-found=true cronjob/env-canada-rdps-${NAME_APP}-${SUFFIX}"
+OC_CLEAN_DEPLOY="oc -n ${PROJ_TARGET} ${DELETE_OR_GET} all,cm,pvc -o name -l app=${NAME_OBJ}"
```
Comment (andrea-williams) [inline openshift/scripts/oc_cleanup.sh:31]: don't we still need the `--ignore-not-found` flag?

File: openshift/certbot/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# cat Dockerfile | oc new-build -D - --name=certbot
```
Comment (conbrad) [inline openshift/certbot/docker/Dockerfile:1]: Minor: commented out command

File: openshift/certbot/openshift/certbot.bc.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+apiVersion: v1
```
Comment (conbrad) [inline openshift/certbot/openshift/certbot.bc.yaml:2]: Should this be `template.openshift.io/v1`?

**Review by andrea-williams — APPROVED** [review]

File: openshift/certbot/openshift/certbot.dc.yaml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+      schedule: "${CERTBOT_CRON_SCHEDULE}"
+      suspend: "${{CERTBOT_SUSPEND_CRON}}"
+
+parameters:
+  # https://crontab.guru/every-6-hours
+  - name: "CERTBOT_CRON_SCHEDULE"
+    required: false
+    value: "0 */12 * * *"
+    # Let's Encrypt Stage environment will help with non-prod cert renewals, so you don't hit limits
+  - name: "CERTBOT_STAGING"
```
Comment (conbrad) [inline openshift/certbot/openshift/certbot.dc.yaml:95]: Does staging work with certbot?

**Review by conbrad — APPROVED** [review]
LGTM, just some minor comments.

File: openshift/Jenkinsfile (hunk 1 above)
Comment (Sybrand) [reply openshift/Jenkinsfile:64]: As far as I know it's not a secret - I'll ask.

File: openshift/scripts/oc_cleanup.sh (hunk 2 above)
Comment (Sybrand) [reply openshift/scripts/oc_cleanup.sh:31]: I think --ignore-not-found is only an issue on cronjobs.

File: openshift/certbot/docker/Dockerfile (hunk 3 above)
Comment (Sybrand) [reply openshift/certbot/docker/Dockerfile:1]: This is taken directly from the certbot project - so I'd rather not change it if I don't need to.

File: openshift/certbot/openshift/certbot.bc.yaml (hunk 4 above)
Comment (Sybrand) [reply openshift/certbot/openshift/certbot.bc.yaml:2]: Again - it being directly from the certbot project I'd rather leave it alone. The build config is only run once, manually - so if it has that warning it doesn't affect us.

**Pull request merged**